api/astro-date.ts

```
@@ -0,0 +1,132 @@
+
+    // Get astrological data
+    console.log('Calculating moon phase...');
+    const moonPhase = getMoonPhase(targetDate);
+    
+    console.log('Calculating planet positions...');
+    const positions = getPlanetPositions(targetDate) as PlanetPositions;
+    
+    // Get zodiac signs for each planet
+    const signs = Object.entries(positions).reduce<Record<string, string>>((acc, [planet, data]) => {
+      return {
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)